sdk/resourcemanager/azure-resourcemanager-cdn/src/main/java/com/azure/resourcemanager/cdn/implementation/CdnEndpointImpl.java

```diff
@@ -72,6 +84,15 @@ public String id() {
     @Override
     public Mono<CdnEndpoint> createResourceAsync() {
         final CdnEndpointImpl self = this;
+        if (isStandardMicrosoftSku()
```

Comment:
I'm afraid only Standard_Microsoft supports Standard rules engine.

Premium has its own rules engine. Both from portal request URLs and from [official docs](https://learn.microsoft.com/en-us/azure/cdn/cdn-verizon-premium-rules-engine) indicate it's a complete new interface. It also involves staging and deploy which standard does not have.

Standard Verizon and Standard Akamai doesn't have Standard rules engine. 
As far as I know they only supports Caching rules , in a separate tab in Portal. It shares the same properties with Standard rules engine rules(`deliveryPolicy`), but each rule's name property must be left blank, otherwise service will reject it, which means `defineNewStandardRulesEngineRule(name)` doesn't work in this case. We may need a separate interface for Caching.

P.S. Comparison between different SKUs: https://learn.microsoft.com/en-us/azure/cdn/cdn-features
<img width="860" alt="Screenshot 2023-03-30 at 16 31 03" src="https://user-images.githubusercontent.com/92354331/228777628-1198ba61-a4fc-462d-978b-e919f48d264b.png">